### PR TITLE
Add detail description of token generation and usage

### PR DESCRIPTION
**Pull request by mcd500**
This PR is inspired of old discussion at mailing list at
https://mailarchive.ietf.org/arch/msg/teep/8BpftgzT6JsocmJtU_bIrY125ek/

It adds explanation of details on generation and usage of token.

@dthaler, please review

### CODE REVIEW

File: draft-ietf-teep-protocol.md [hunk 1]
```diff
@@ -326,8 +326,16 @@ type
   the TEEP Agent.
 
 token
-: The value in the token parameter is used to match responses to requests. This is 
-particularly useful when a TAM issues multiple concurrent requests to a TEEP Agent. The token is not needed when the attestation bit is set in the data-item-requested value.
+: The value in the token parameter is used to match responses to requests.
+  This is particularly useful when a TAM issues multiple concurrent requests
+  to a TEEP Agent. The token is not needed when the attestation bit is set in
+  the data-item-requested value. The size of the token is at least 8 bytes
+  (64 bits) and maximum of 64 bytes, which is the same as Nonce Claim in EAT.
```
Comment (dthaler) [inline draft-ietf-teep-protocol.md:333]:
```suggestion
  (64 bits) and maximum of 64 bytes, which is the same as in an EAT Nonce Claim
```

Comment (mcd500) [reply draft-ietf-teep-protocol.md]: @dthaler thanks, pushed the change

File: draft-ietf-teep-protocol.md [hunk 2]
```diff
@@ -326,8 +326,16 @@ type
...
+  (see {{I-D.ietf-rats-eat}} Section 3.3). The first usage of a token
+  generated by TAM MUST be randomly created so the device will unlikely
+  receive the same token value when the device handles multiple TAMs.
+  Consequent token value MUST be a different value for each request message
+  to distinguish the correct response from multiple requests.
```
[the remark this answers is not in the document]
Comment (mcd500) [reply draft-ietf-teep-protocol.md]: @dthaler also pushed the change

File: draft-ietf-teep-protocol.md [hunk 3]
```diff
@@ -326,8 +326,15 @@ type
...
 
 token
-: The value in the token parameter is used to match responses to requests. This is 
-particularly useful when a TAM issues multiple concurrent requests to a TEEP Agent. The token is not needed when the attestation bit is set in the data-item-requested value.
+: The value in the token parameter is used to match responses to requests.
+  This is particularly useful when a TAM issues multiple concurrent requests
+  to a TEEP Agent. The token is not needed when the attestation bit is set in
+  the data-item-requested value. The size of the token is at least 8 bytes
+  (64 bits) and maximum of 64 bytes, which is the same as in an EAT Nonce
+  Claim. (see {{I-D.ietf-rats-eat}} Section 3.3). The first usage of a token
```
Comment (dthaler) [inline draft-ietf-teep-protocol.md:334]:
```suggestion
  Claim (see {{I-D.ietf-rats-eat}} Section 3.3). The first usage of a token
```

File: draft-ietf-teep-protocol.md [hunk 4]
```diff
@@ -326,8 +326,15 @@ type
...
+  generated by TAM MUST be randomly created.
```
Comment (dthaler) [inline draft-ietf-teep-protocol.md:335]:
```suggestion
  generated by a TAM MUST be randomly created.
```

File: draft-ietf-teep-protocol.md [hunk 5]
```diff
@@ -326,8 +326,15 @@ type
...
+  Consequent token value MUST be a different value for each request message
```
Comment (dthaler) [inline draft-ietf-teep-protocol.md:336]:
```suggestion
  Consequent token values MUST be different for each request message
```

**Comment by dthaler** [comment]
Ok except for grammar nits. Ok to merge if my suggestions are accepted.

**Comment by mcd500** [comment]
@dthaler thanks for your review, merged

**Pull request merged**